### PR TITLE
Violation type and violation classification admin services

### PR DESCRIPTION
**Pull request by ModarEskandar**

### CODE REVIEW

File: src/modules/administration/components/localization/localization.component.html [hunk 1]
```diff
@@ -14,7 +14,8 @@ <h1 class="text-xl min-h-[3rem] align-middle flex items-center">
         <mat-checkbox
           (change)="$event ? toggleAllRows() : null"
           [checked]="selection.hasValue() && isAllSelected()"
-          [indeterminate]="selection.hasValue() && !isAllSelected()">
+          [indeterminate]="selection.hasValue() && !isAllSelected()"
+          #c>
```
Comment (BeSaRa) [inline src/modules/administration/components/localization/localization.component.html:18]: why you add this template reference ??

File: src/modules/administration/components/violation-classification/violation-classification.component.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
+<div
+  class="flex bg-gradient-to-br from-primary to-primary-light text-white rounded-tr-lg rounded-tl-lg items-center justify-start py-2">
+  <h1 class="text-xl min-h-[3rem] align-middle flex items-center px-3">
+    {{ lang.map.menu_violation_classification }}
+  </h1>
+  <app-icon-button [matTooltip]="lang.map.reload" (click)="reload$.next()" icon="RELOAD"></app-icon-button>
+  <app-icon-button [matTooltip]="lang.map.add" (click)="create$.next()" icon="PLUS"></app-icon-button>
+  <div class="flex-auto"></div>
+  <div>
+    <app-icon-button icon="DELETE" [matTooltip]="lang.map.delete_bulk" z></app-icon-button>
```
Comment (BeSaRa) [inline src/modules/administration/components/violation-classification/violation-classification.component.html:10]: what is "z" attribute ?

File: src/modules/administration/components/localization/localization.component.html (hunk 1 above)
Comment (ModarEskandar) [reply src/modules/administration/components/localization/localization.component.html:18]: when all selected and reload this checkbox still selected but all rows became unselected
try to solve this issue

File: src/modules/administration/components/violation-classification/violation-classification.component.html (hunk 2 above)
Comment (ModarEskandar) [reply src/modules/administration/components/violation-classification/violation-classification.component.html:10]: leaked from ctrl+z i suppose

**Pull request closed without merging**